secure/storage/src/namespaced_storage.rs

```diff
@@ -33,20 +65,59 @@ impl KVStorage for NamespacedStorage {
     }
 }
 
-impl NamespacedStorage {
-    pub fn new(storage: Storage, namespace: String) -> Self {
-        NamespacedStorage {
-            namespace,
-            inner: Box::new(storage),
-        }
+impl<S: CryptoStorage> CryptoStorage for Namespaced<S> {
```

Comment:
Hmm.. all of this boilerplate isn't super great. What stops us from using/modifying the default `CryptoKVStorage` implementation?

Reply:
I disagree that this boilerplate isn't great. Its actually more accurate than the old Namespaced implementation since this will forward the calls to the underlying CryptoStorage implementation while the only one wouldn't. This is required in order to allow for Vault to be wrapped by Namespaced

Reply:
For what it's worth, there should be no other CryptoStorage implementations today (other than the default) so we shouldn't be too concerned about wanting to forward them to the "underlying" implementation, but I see your point. Likewise, my comment about boilerplate is that generally I'm not a big fan. 😄